lua/weapons/weapon_laserpointer.lua

```diff
@@ -0,0 +1,326 @@
+
+		if(mrt)then
+			mrt:SetTranslation(opos)
+			mrt:SetAngles(oang)
+			self:SetBoneMatrix(0, mrt)
+		end
+	end
+	self:DrawModel()
+end
+
+function SWEP:GetViewModelPosition(pos, ang)
+	pos = pos + ang:Right()*2
+	pos = pos + ang:Up()*-3
+	pos = pos + ang:Forward()*5
+	ang:RotateAroundAxis(ang:Forward(), -3)
+	ang:RotateAroundAxis(ang:Right(), -90)
+	ang:RotateAroundAxis(ang:Up(), -105)
+	pos = pos + ang:Forward()*0.1
+	return pos, ang
+end
+
+function SWEP:PreDrawViewModel()
+	if IsValid(self) and self:GetClass() == "weapon_laserpointer" and self.LaserBeamOn and self.Weapon:Clip1() > 0  then
+		local tr = self.Owner:GetEyeTrace()
+		local vm = self.Owner:GetViewModel()
+		local attachmentIndex = vm:LookupAttachment("tip")
+		local origin = vm:GetAttachment(attachmentIndex).Pos
+		local lsrclr = self.LaserColors[self.LaserPower+1]
+
+		cam.Start3D(EyePos(), EyeAngles())
+			local size = math.random()*10
+
+			render.SetMaterial(Material("sprites/bluelaser1"))
+			render.DrawBeam(origin, tr.HitPos, 2, 0, 12.5, lsrclr, false)
+			render.SetMaterial(Material("sprites/light_glow02_add_noz"))
+			render.DrawSprite(tr.HitPos, size, size, lsrclr)
+		cam.End3D()
+		--[[if tr.Entity:GetMaterial() == "func_reflective_glass" and maxbounce <= 1 then--for reflecting the laser
+			local vec = origin-tr.HitPos
+			local normal = tr.HitNormal
+			local reflect = -2*vec:Dot(normal)*normal+vec
+			local trc = util.TraceLine({
+				start = tr.HitPos,
+				endpos = tr.HitPos + reflect:Forward() * 10000,
+				filter = false
+			})
+
+			render.SetMaterial(Material("sprites/bluelaser1"))
+			render.DrawBeam(tr.HitPos, trc.HitPos, 2, 0, 12.5, lsrclr, false)
+			render.SetMaterial(Material("sprites/light_glow02_add_noz"))
+			render.DrawSprite(trc.HitPos, size, size, lsrclr)
+
+			maxbounce = maxbounce+1
+		end]]
+	end
+end
+
+function SWEP:Deploy()
+	self:RemoveLaser()
+	self:SetHoldType("pistol")
+    self.Weapon:SendWeaponAnim(ACT_VM_DRAW)
+end
+
+function SWEP:OnRemove()
+	self:RemoveLaser()
+end
+
+function SWEP:PrimaryAttack()--shoots the laser
+	if (!self:CanPrimaryAttack()) then return end
+
+	local trc = self.Owner:GetEyeTrace()
+	local lctn = Location.GetLocationNameByIndex(Location.Find(trc.Entity)):lower()
+	local loc = Location.GetLocationNameByIndex(Location.Find(self.Owner)):lower()
+
+	if (self.Owner:InTheater() and not (self.Owner:GetTheater()._AllowItems)) or loc=="trump lobby" or loc=="golf" then
+		self.Owner:PrintMessage(HUD_PRINTTALK, "[red] Stop right there! Lasers are not allowed in this area. ;authority;")
+	end
+
+	self:DrawLaser()
+	self.Weapon:TakePrimaryAmmo(self.LaserPower/2+0.1)
+
+	if IsValid(trc.Entity) and trc.Entity:IsPlayer() and trc.HitBox == 0 and not -Safe(trc.Entity) and (trc.Entity:InTheater() and not (trc.Entity:GetTheater()._AllowItems)) or lctn=="trump lobby" or lctn=="golf" then
+		self:BlindPlayer(trc.Entity)
+		self:SetNextPrimaryFire(CurTime() + 0.1)
+	else
+		self:SetNextPrimaryFire(CurTime() + 0.1)

```

Comment:
factor out of conditional

Reply:
does this mean you want me to move the SetNextPrimaryFire out of the conditional?

Reply:
yeah there is no reason to have it on both side of if/else, just put it at the bottom

Reply:
Yep already done